dd-trace-core/src/main/java/datadog/trace/core/datastreams/DefaultPathwayContext.java

```diff
@@ -65,46 +68,25 @@ public boolean isStarted() {
   }
 
   @Override
-  public void start(Consumer<StatsPoint> pointConsumer) {
-    setCheckpoint(null, null, INITIALIZATION_TOPIC, pointConsumer);
-  }
-
-  @Override
-  public void setCheckpoint(
-      String type, String group, String topic, Consumer<StatsPoint> pointConsumer) {
-
+  public void setCheckpoint(List<String> tags, Consumer<StatsPoint> pointConsumer) {
     long startNanos = timeSource.getCurrentTimeNanos();
     long nanoTicks = timeSource.getNanoTicks();
-
     lock.lock();
     try {
-      if (INITIALIZATION_TOPIC.equals(topic) && started) {
-        return;
-      }
-
       List<String> edgeTags = new ArrayList<>();
 
-      if (started) {
-        // Only create edge tags if there's a parent (ie context has started)
-        if (type != null && !type.isEmpty()) {
-          edgeTags.add("type:" + type);
-        }
-
-        if (topic != null && !topic.isEmpty()) {
-          edgeTags.add("topic:" + topic);
-        }
-
-        if (group != null && !group.isEmpty()) {
-          edgeTags.add("group:" + group);
-        }
-      } else {
+      if (!started) {
         pathwayStartNanos = startNanos;
         pathwayStartNanoTicks = nanoTicks;
         edgeStartNanoTicks = nanoTicks;
         hash = 0;
         started = true;
         log.debug("Started {}", this);
       }
+      if (tags != null) {
+        // Only create edge tags if there's a parent (ie context has started)
+        edgeTags.addAll(tags);
+      }
```

Comment:
Is there a valid case where `tags` is `null`? I think it should be an empty list if there are no tags with `null` being invalid.

```suggestion
      edgeTags.addAll(tags);
```